salt/orch/force_removal.sls

```diff
@@ -0,0 +1,72 @@
+# This is a best effort orchestration. Any operation run on the minions by this orchestration
+# could fail, as we don't know the current state of the cluster at this point. The idea is to
+# try to clean as much as possible where possible, but in the end the only thing we can really
+# ensure is that the salt-key of the target is rejected.
+
+# must provide the node (id) to be removed in the 'target' pillar
+{%- set target = salt['pillar.get']('target') %}
+{%- set target_nodename = salt.saltutil.runner('mine.get', tgt=target, fun='nodename')[target] %}
+# services that will be called when trying to unregister as clean as possible
+{%- set services_to_stop = { "kubelet": ["stop", "remove-pre-reboot"], "etcd": ["remove-pre-stop-services", "remove-pre-reboot"] } %}
+# nodes that are responding
+{%- set up_nodes = salt.saltutil.runner('manage.up', tgt='P@roles:kube-* and not ' + target, expr_form='compound') %}
+
+# Try to unregister the machine from cluster services. It's important to keep every sls in
+# different toplevel stages, so each one will be tried even if previous fails for whatever
+# reason.
+{%- for service, sls_list in services_to_stop.items() %}
+  {%- for sls in sls_list %}
+stop-{{ service }}-{{ sls }}-stage:
+  salt.state:
+    - tgt: {{ target }}
+    - fail_minions: {{ target }}
+    - sls:
+        - {{ service }}.{{ sls }}
+  {%- endfor %}
+{%- endfor %}
```

Comment:
I think you need to skip this whole block if `target in nodes_down`. Otherwise, the `salt.state` will fail (even with `fail_minions`) as the Salt minion is not responding...